paddle/fluid/platform/mkldnn_reuse.h

```diff
@@ -828,12 +828,16 @@ class TransposeMKLDNNHandler : public MKLDNNHandler {
  public:
   TransposeMKLDNNHandler(std::vector<int>& dims,  // NOLINT
```

Comment:
It'd be much better to tempalte parametrise TransposeMKLDNNHandler. That way we avoid passing used types (vtype and dtype) as arguments during runtime, as well as avoid cache trashing by adding new fields to the class.

```C++
template <typename T>
TransposeMKLDNNHandler(std::vector<int>& dims,  // NOLINT
                         std::vector<int>& axis,  // NOLINT
                         const platform::MKLDNNDeviceContext& dev_ctx,
                         mkldnn::engine engine, const std::string& base_key)
      : platform::MKLDNNHandler(dev_ctx, engine, base_key),
        dims_(dims),
        axis_(axis),
        logical_axis_(dims.size(), 0) {}
```
I'll continue describing necessary changes in comments that follow.

Reply:
I've done it.